server.js

```diff
@@ -26,7 +57,102 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("Welcome to top music on Spotify! Look for your favorite top music with these endpoints.");
+});
+
+app.get("/songs/id/:id", async (req, res) => {
+  try {
+    const singleSong = await Song.findById(req.params.id)
+    if (singleSong) {
+      res.status(200).json({
+        success: true,
+        body: singleSong
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "Could not find the song"
+        }
+      });
+    }
+  } catch(error) {
+    res.status(400).json({
+      success: false,
+      body: {
+        message: "Invalid id"
+      }
+    });
+  }
+});
+
+app.get("/songs/", async (req, res) => {
+
+  const {
+    genre, 
+    trackName, 
+    artistName,
+    bpm, 
+    energy,
+    danceability,
+    loudness,
+    liveness,
+    valence,
+    length,
+    acousticness,
+    speechiness,
+    popularity
+  } = req.query;
+  
+  const response = {
+    success: true,
+    body: {}
+  }
+
+  const trackNameQuery = trackName ? trackName : /.*/;
+  const genreQuery = genre ? genre : /.*/;
+  const artistNameQuery = artistName ? artistName : /.*/;
+  const bpmQuery = bpm ? bpm : {$gt: 0, $lt: 300};
+  const energyQuery = energy ? energy : {$gt: 0, $lt: 100};
+  const danceabilityQuery = danceability ? danceability : {$gt: 0, $lt: 100};
+  const loudnessQuery = loudness ? loudness : {$gt: -100, $lt: 1};
+  const livenessQuery = liveness ? liveness : {$gt: 0, $lt: 100};
+  const valenceQuery = valence ? valence : {$gt: 0, $lt: 100};
+  const lengthQuery = length ? length : {$gt: 0, $lt: 1000};
+  const acousticnessQuery = acousticness ? acousticness : {$gt: 0, $lt: 100};
+  const speechinessQuery = speechiness ? speechiness : {$gt: 0, $lt: 100};
+  const popularityQuery = popularity ? popularity : {$gt: 0, $lt: 100};
```

Comment:
Great job with providing the values with regex and values greater than/lower than.